### PR TITLE
GC Sweep Tests: Enable inactive object x errors and file logging

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#1797](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1797)

Enable InactiveObject_X errors and log those errors and other information to a few files for easier debugging.

### CODE REVIEW

File: packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts [hunk 1]
```diff
@@ -74,12 +75,14 @@ describeNoCompat("GC Sweep tests", (getTestObjectProvider) => {
...
-    const sessionExpiryDurationMs = 3000; // 3 seconds
+    // const sessionExpiryDurationMs = 3000; // 3 seconds
+    const inactiveTimeoutMs = 3000; // 3 seconds
 
     // Set settings here, may be useful to put everything in the mockConfigProvider
     const settings = {
-        "Fluid.GarbageCollection.RunSessionExpiry": "true",
-        "Fluid.GarbageCollection.TestOverride.SessionExpiryMs": sessionExpiryDurationMs,
+        // "Fluid.GarbageCollection.RunSessionExpiry": "true",
+        "Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs": inactiveTimeoutMs,
```
Comment (tyler-cai-microsoft) [inline packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:84]: After enabling this, I noticed that my tests would only run once. After removing it, the test would run 10 times.

File: packages/test/test-gc-sweep-tests/src/ignoreErrorLogger.ts [hunk 2]
```diff
@@ -13,6 +12,10 @@ import { ITelemetryBaseEvent, ITelemetryGenericEvent } from "@fluidframework/com
  */
 export class IgnoreErrorLogger extends EventAndErrorTrackingLogger {
     private readonly ignoredEvents: Map<string, ITelemetryGenericEvent> = new Map();
```
[the remark this answers is not in the document]
Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/ignoreErrorLogger.ts:14]: No, I'll delete it

File: packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts (hunk 1 above)
Comment (markfields) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:83]: This sounds like a bug, maybe open a work item in ADO?

Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:83]: AB#1844 created

File: packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts [hunk 3]
```diff
@@ -267,33 +263,44 @@ describeNoCompat("GC Sweep tests", (getTestObjectProvider) => {
...
             };
 
+            fs.mkdirSync(`nyc/testData-${seed}`, { recursive: true });
+            fs.writeFileSync(`nyc/testData-${seed}/events.json`, JSON.stringify(overrideLogger.events));
+            fs.writeFileSync(`nyc/testData-${seed}/inactiveObjectEvents.json`, JSON.stringify(overrideLogger.inactiveObjectEvents));
+            fs.writeFileSync(`nyc/testData-${seed}/errorEvents.json`, JSON.stringify(overrideLogger.errorEvents));
+            fs.writeFileSync(`nyc/testData-${seed}/errorEventStats.json`, JSON.stringify(overrideLogger.errorEventStats));
+            fs.writeFileSync(`nyc/testData-${seed}/actions.json`, JSON.stringify(actionsList));
+            fs.writeFileSync(`nyc/testData-${seed}/stats.json`, JSON.stringify(stats));
+            fs.writeFileSync(`nyc/testData-${seed}/errors.json`, JSON.stringify(errorList));
```
Comment (markfields) [inline packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:283]: Might be nice to wrap some of these in an uber-object and log that to a single file?  Up to you, I haven't actually worked with the process/data.

File: packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts [hunk 4]
```diff
@@ -267,33 +263,44 @@ describeNoCompat("GC Sweep tests", (getTestObjectProvider) => {
...
+                ...actionStats,
                 actionCount: actionsList.length,
-                actionsList,
                 errorCount: errorList.length,
-                errorList,
-                ...actionStats,
-                handleRecord: handleTracker.handleActionRecord,
             };
 
+            fs.mkdirSync(`nyc/testData-${seed}`, { recursive: true });
```
Comment (markfields) [inline packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:276]: Why nyc?  I know that's used for other test output in some cases but does it apply here?  I think that's based on the name of the package that is used or something.  No harm in using the name if it doesn't apply, just curious.

Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:276]: It's just so when you run the tests, that it's ignored for git.

Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:276]: If this test gets fleshed out more, we should be using something that would report to the nyc folder most likely.

**Review by markfields — APPROVED** [review]
🎂 LGTM!

File: packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts [hunk 5]
```diff
@@ -74,12 +74,14 @@ describeNoCompat("GC Sweep tests", (getTestObjectProvider) => {
         runtimeOptions,
     );
 
-    const sessionExpiryDurationMs = 3000; // 3 seconds
+    // const sessionExpiryDurationMs = 3000; // 3 seconds
+    const inactiveTimeoutMs = 3000; // 3 seconds
```
Comment (agarwal-navin) [inline packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:78]: Shouldn't we make this configurable via test configs or something?

Comment (agarwal-navin) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:78]: Why do we need a default value here?

Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:78]: Yeah, there is no current configurable options, everything is hard coded until we add that functionality.

Comment (tyler-cai-microsoft) [reply packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts:78]: I'll add this to the next PR as this seems like a decent sized chunk to look through.

**Pull request merged**